packages/firestore/src/core/sync_engine.ts

```diff
@@ -1371,3 +1380,57 @@ export function newMultiTabSyncEngine(
     maxConcurrentLimboResolutions
   );
 }
+
+export function loadBundle(
+  syncEngine: SyncEngine,
+  bundleReader: BundleReader,
+  task: LoadBundleTaskImpl
+): Promise<void> {
```

Comment:
Did you consider returning the LoadBundleTask instead of having it passed in? You are now mixing two async patterns in one callsite.